ArduCopter/sensors.cpp

```diff
@@ -93,7 +93,8 @@ void Copter::init_compass()
         return;
     }
 
-    if (!compass.init() || !compass.read()) {
+    compass.init();
+    if (!compass.read()) {
         // make sure we don't pass a broken compass to DCM
         hal.console->printf("COMPASS INIT ERROR\n");
         Log_Write_Error(ERROR_SUBSYSTEM_COMPASS,ERROR_CODE_FAILED_TO_INITIALISE);
```

Comment:
Applies to all vehicles: if compass init never fails, these two seem a bit strange (more now that it's when doing a read).